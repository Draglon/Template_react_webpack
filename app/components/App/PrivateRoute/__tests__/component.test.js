import React from 'react';
import { BrowserRouter as Router, Route, Redirect } from 'react-router-dom';

import PrivateRouteComponent from '../component';

describe('<PrivateRouteComponent />', () => {
  const location = '/dashboard';
  const props = {
    component: () => <h1>Hyperspace tracker</h1>,
    path: location,
    sessionId: '',
  };

  const wrapper = shallow(<PrivateRouteComponent {...props} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Without sessionId', () => {
    beforeEach(() => {
      wrapper.setProps(props);
    });

    it('should check redirect path', () => {
      const rendered = wrapper.renderProp('render')({ location });
      expect(rendered.find(Redirect).props()).toEqual({
        push: false,
        to: {
          pathname: '/',
          state: { from: location },
        },
      });
    });

    it('should match snapshot', () => {
      const rendered = wrapper.renderProp('render')({ location });
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('With sessionId', () => {
    const newProps = {
      ...props,
      sessionId: 'some sessionId',
    };

    beforeEach(() => {
      wrapper.setProps(newProps);
    });

    it('should match snapshot', () => {
      const rendered = wrapper.renderProp('render')();
      expect(rendered).toMatchSnapshot();
    });
  });
});
