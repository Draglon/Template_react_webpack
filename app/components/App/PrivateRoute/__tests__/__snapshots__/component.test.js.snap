// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRouteComponent /> With sessionId should match snapshot 1`] = `<component />`;

exports[`<PrivateRouteComponent /> Without sessionId should match snapshot 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/",
      "state": Object {
        "from": "/dashboard",
      },
    }
  }
/>
`;

exports[`<PrivateRouteComponent /> should match snapshot 1`] = `
<Route
  path="/dashboard"
  render={[Function]}
/>
`;
